config/src/test/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurerTests.java

```diff
@@ -0,0 +1,259 @@
+		this.response = new MockHttpServletResponse();
+		this.filterChain = new MockFilterChain();
+	}
+
+	@After
+	public void cleanup() {
+		if (this.context != null) {
+			this.context.close();
+		}
+	}
+
+	@Test
+	public void saml2LoginWhenConfiguringAuthenticationManagerThenTheManagerIsUsed() throws Exception {
+		// setup application context
+		this.spring.register(Saml2LoginConfigWithCustomAuthenticationManager.class).autowire();
+		performSaml2Login("ROLE_AUTH_MANAGER");
+	}
+
+	@Test
+	public void saml2LoginWhenConfiguringAuthenticationDefaultsUsingCustomizerThenTheProviderIsConfigured() throws Exception {
+		// setup application context
+		this.spring.register(Saml2LoginConfigWithAuthenticationDefaultsWithPostProcessor.class).autowire();
+		validateSaml2WebSsoAuthenticationFilterConfiguration();
+	}
+
+	private void validateSaml2WebSsoAuthenticationFilterConfiguration() {
+		// get the OpenSamlAuthenticationProvider
+		Saml2WebSsoAuthenticationFilter filter = getSaml2SsoFilter(this.springSecurityFilterChain);
+		AuthenticationManager manager =
+				(AuthenticationManager) ReflectionTestUtils.getField(filter, "authenticationManager");
+		ProviderManager pm = (ProviderManager) manager;
+		AuthenticationProvider provider = pm.getProviders()
+				.stream()
+				.filter(p -> p instanceof OpenSamlAuthenticationProvider)
+				.findFirst()
+				.get();
+		Assert.assertSame(AUTHORITIES_EXTRACTOR, ReflectionTestUtils.getField(provider, "authoritiesExtractor"));
+		Assert.assertSame(AUTHORITIES_MAPPER, ReflectionTestUtils.getField(provider, "authoritiesMapper"));
+		Assert.assertSame(RESPONSE_TIME_VALIDATION_SKEW, ReflectionTestUtils.getField(provider, "responseTimeValidationSkew"));
+	}
+
+	private Saml2WebSsoAuthenticationFilter getSaml2SsoFilter(FilterChainProxy chain) {
+		return (Saml2WebSsoAuthenticationFilter) chain.getFilters("/login/saml2/sso/test")
+				.stream()
+				.filter(f -> f instanceof Saml2WebSsoAuthenticationFilter)
+				.findFirst()
+				.get();
+	}
+
+	private void performSaml2Login(String expected) throws IOException, ServletException {
+		// setup authentication parameters
+		this.request.setParameter(
+				"SAMLResponse",
+				Base64.getEncoder().encodeToString(
+						"saml2-xml-response-object".getBytes()
+				)
+		);
+
+
+		// perform test
+		this.springSecurityFilterChain.doFilter(this.request, this.response, this.filterChain);
+
+		// assertions
+		Authentication authentication = this.securityContextRepository
+				.loadContext(new HttpRequestResponseHolder(this.request, this.response))
+				.getAuthentication();
+		Assert.assertNotNull("Expected a valid authentication object.", authentication);
+		assertThat(authentication.getAuthorities()).hasSize(1);
+		assertThat(authentication.getAuthorities()).first()
+				.isInstanceOf(SimpleGrantedAuthority.class).hasToString(expected);
+	}
+
+
+	@EnableWebSecurity
+	static class Saml2LoginConfigWithCustomAuthenticationManager extends Saml2LoginConfigBase {
```

Comment:
Would

```suggestion
	@Import(Saml2LoginConfigBase.class)
	static class Saml2LoginConfigWithCustomAuthenticationManager extends WebSecurityConfigurerAdapter {
```

be a bit simpler? I guess it feels a little odd to extend `WebSecurityConfigurerAdapter` when only adding beans into the application context, it seems like it's doing two jobs.

Reply:
Agreed